### PR TITLE
add item #191

### LINKED ISSUE #191: Implement User Interface and API request for adding items
- [ ] Items button/link on the landing page and the table view 
- [ ] Create back button for the landing page and the table view 
- [ ] Implement Add item button and dialog.
- [ ] the add items dialog should have pre-populated values from the catalogue item

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description

A user should be able to add an item. A user should also be able to navigate to the items page on the catalogue category table view and the catalogue item landing page. The items page should allow the user to navigate back to either of the pages.  

- ~The items dialog unit tests and the items e2e test  fail locally due the Timezone issues. Ignore this for now. We will deal with this after MVP.~ - Fixed
-  
## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking

closes #191

### CODE REVIEW

**Review by joelvdavies — CHANGES_REQUESTED** [review]
Mostly just me being pedantic, would be good if the testing issue could fixed though. Worked well when I tested it locally, also checked the behaviour of changing my local time zone which atomically converted the times on the front end, which I think is reasonable. The backend just remains in UTC.

File: src/items/itemDialog.component.tsx [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+      asset_number: itemDetails.asset_number,
+      serial_number: itemDetails.serial_number,
+      delivered_date:
+        itemDetails.delivered_date &&
+        isValidDateTime(itemDetails.delivered_date)
+          ? new Date(itemDetails.delivered_date).toISOString() ?? null
+          : null,
+      notes: itemDetails.notes,
+    };
+  }, [itemDetails, catalogueItem]);
```
[the remark this answers is not in the document]
Comment (joshuadkitenge) [reply src/items/itemDialog.component.tsx:303]: This is going to be used for the handleEditItem as well. I'll add the type on it to make it clearer what it is

[the remark this answers is not in the document]
Comment (joelvdavies) [reply src/items/itemDialog.component.tsx]: Shouldn't edit use undefined instead though?

File: src/items/items.component.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        width: '100%',
+      }}
+    >
+      <Button
+        sx={{ mx: 0.5 }}
+        component={Link}
+        startIcon={
+          <Box sx={{ alignItems: 'center', display: 'flex' }}>
+            <ArrowBackIcon fontSize="small" />{' '}
+            <ArrowBackIcon fontSize="small" />
```
Comment (MatteoGuarnaccia5) [inline src/items/items.component.tsx:38]: image
Double arrow makes it look like this. Shouldn't it be a single arrow?

Comment (joshuadkitenge) [reply src/items/items.component.tsx:38]: It meant to be a double as you are going back up two levels

File: src/items/itemDialog.component.tsx [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+          }}
+        >
+          <Button
+            variant="outlined"
+            sx={{ width: '50%', mx: 1 }}
+            onClick={handleClose}
+          >
+            Cancel
+          </Button>
+          <Button
```
Comment (MatteoGuarnaccia5) [inline src/items/itemDialog.component.tsx:614]: add logic so save button gets disabled when there is an error in the dialog. see #189

**Review by MatteoGuarnaccia5 — COMMENTED** [review]
Just a couple of things from me (see above)

**Review by MatteoGuarnaccia5 — APPROVED** [review]
LGTM

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**